pkg/traces/exporter/exporter.go

```diff
@@ -53,37 +72,50 @@ func NewTraceExporter(ctx context.Context, serverProvidedDataStore types.Getter,
 		attrs = append(attrs, archAttr)
 	}
 
+	currentToken, _ := k.ConfigStore().Get([]byte(configStoreTokenKey))
+
 	t := &TraceExporter{
-		serverProvidedDataStore: serverProvidedDataStore,
-		osqueryClient:           client,
-		attrs:                   attrs,
-		attrLock:                sync.RWMutex{},
+		knapsack:        k,
+		osqueryClient:   client,
+		logger:          log.With(logger, "component", "trace_exporter"),
+		attrs:           attrs,
+		attrLock:        sync.RWMutex{},
+		ingestAuthToken: string(currentToken),
+		ingestUrl:       k.IngestServerURL(),
+		enabled:         k.ExportTraces(),
 	}
 
-	t.addDeviceIdentifyingAttributes()
+	// Observe ExportTraces and IngestServerURL changes to know when to start/stop exporting, and where
+	// to export to
+	t.knapsack.RegisterChangeObserver(t, keys.ExportTraces, keys.IngestServerURL)
 
-	// Set the provider with as many resource attributes as we could get immediately
-	exp, err := newExporter(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("could not create new exporter: %w", err)
+	if !t.enabled {
+		return t, nil
 	}
-	t.setNewGlobalProvider(exp)
+
+	t.addDeviceIdentifyingAttributes()
+
+	// Set the provider with as many resource attributes as we can get immediately
+	t.setNewGlobalProvider()
 
 	// In the background, wait for osquery to be ready so that we can fetch more resource
 	// attributes for our traces, then replace the provider with a new one.
 	go func() {
 		t.addAttributesFromOsquery()
-		if exp, err := newExporter(ctx); err == nil {
-			t.setNewGlobalProvider(exp)
-		}
+		t.setNewGlobalProvider()
+		level.Debug(t.logger).Log("msg", "successfully replaced global provider after adding osquery attributes")
 	}()
 
 	return t, nil
 }
 
-// newExporter returns an exporter that will send traces with OTLP over HTTP.
-func newExporter(ctx context.Context) (sdktrace.SpanExporter, error) {
-	traceClient := otlptracegrpc.NewClient(otlptracegrpc.WithInsecure())
+// newExporter returns an exporter that will send traces with OTLP over gRPC.
+func newExporter(ctx context.Context, token string, url string) (sdktrace.SpanExporter, error) {
+	traceClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithInsecure(),
```

Comment:
do we want to default to `WithInsecure`? or have a flag?

Reply:
We don't want to default to `WithInsecure` ultimately -- I will add a flag, just hadn't gotten to it yet 😅

Reply:
This is added now!